CMakeLists.txt

```diff
@@ -5,6 +5,23 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
 
 Option(DEV_ENVIRONMENT "DEV_ENVIRONMENT" OFF)
 
+#####################################################################
+# decide which components need to be built: by default, build everything,
+# unless some components were explicitely disabled from the parent project
+#####################################################################
+if(NOT DEFINED TERRAPHAST_USE_GMP)
```

Comment:
This would probably be best solved using `Option`, so you don't need to do the default handling yourself

Reply:
can I set option values (or overwrite their default values) from the parent scope? since that's why i added those lines, i want to configure terraphast from my (parent) CMakeLists, like here:

https://github.com/amkozlov/raxml-ng/blob/dev/libs/CMakeLists.txt#L41

Reply:
It seems to work that way: https://stackoverflow.com/questions/20239334/cmake-set-subdirectory-options